addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/WebSocketBase.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.samsungtv.internal.protocol;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketAdapter;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Websocket base class
+ *
+ * @author Arjan Mels - Initial contribution
+ */
+class WebSocketBase extends WebSocketAdapter {
+    private final Logger logger = LoggerFactory.getLogger(WebSocketBase.class);
+    /**
+     *
+     */
+    final RemoteControllerWebSocket remoteControllerWebSocket;
+
+    /**
+     * @param remoteControllerWebSocket
+     */
+    WebSocketBase(RemoteControllerWebSocket remoteControllerWebSocket) {
+        this.remoteControllerWebSocket = remoteControllerWebSocket;
+    }
+
+    boolean isConnecting = false;
+
+    @Override
+    public void onWebSocketClose(int statusCode, String reason) {
+        logger.debug("{} connection closed: {} - {}", this.getClass().getSimpleName(), statusCode, reason);
+        super.onWebSocketClose(statusCode, reason);
+        isConnecting = false;
+    }
+
+    @Override
+    public void onWebSocketError(Throwable error) {
+        logger.warn("{} connection error: {}", this.getClass().getSimpleName(), error.getMessage());
+        super.onWebSocketError(error);
+        isConnecting = false;
+    }
+
+    void connect(URI uri) throws RemoteControllerException {
+        if (isConnecting || isConnected()) {
+            logger.trace("{} already connecting or connected", this.getClass().getSimpleName());
+            return;
+        }
+
+        logger.debug("{} connecting to: {}", this.getClass().getSimpleName(), uri);
+        isConnecting = true;
+
+        try {
+            remoteControllerWebSocket.client.connect(this, uri, new ClientUpgradeRequest());
+        } catch (IOException e) {
+            throw new RemoteControllerException(e);
+        }
+    }
+
+    @Override
+    public void onWebSocketConnect(Session session) {
+        logger.debug("{} connection established: {}", this.getClass().getSimpleName(),
+                session.getRemoteAddress().getHostString());
+        super.onWebSocketConnect(session);
+
+        isConnecting = false;
+    }
+
+    void close() {
+        logger.debug("{} connection close requested", this.getClass().getSimpleName());
+        getSession().close();
+    }
+
+    void sendCommand(String cmd) {
+        try {
+
```

Comment:
Can you run the [static analysis tool](https://www.openhab.org/docs/developer/development/guidelines.html#static-code-analysis) and fix the findings such as removing these unnecessary new lines?

Reply:
Ok, busy with the static code analysis. I think it is usefull, but solving all the @NonNullByDefault stuff will take me a bit of time.

Reply:
I'm OK with not solving all the `@NonNullByDefault` issues. Especially with POJOs that have only nullable fields they are quite laborious. They should have invented a `@NullByDefault` annotation for those. :wink:

Reply:
Yeah, too late, done now ;-)

Reply:
BTW you can use  `@NonNullByDefault({})` as a "`@NullByDefault`" annotation. You can exactly specify which thing the non-null should apply to. [See eclipse java documentation.](https://help.eclipse.org/luna/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjdt%2Fannotation%2FDefaultLocation.html)